openedx/features/edly/tests/test_utils.py

```diff
@@ -346,3 +348,73 @@ def test_create_learner_link_with_permission_groups(self):
+                'company_about_url': 'fake-about-url'
+            }
+        }
+
+        expected_current_site_context_data = test_default_certificate_html_configurations['default']
+        curent_site_context_data = get_current_site_invalid_certificate_context(test_default_certificate_html_configurations)
+
+        assert expected_current_site_context_data['platform_name'] == curent_site_context_data['platform_name']
+        assert expected_current_site_context_data['company_privacy_url'] == curent_site_context_data['company_privacy_url']
+        assert expected_current_site_context_data['company_tos_url'] == curent_site_context_data['company_tos_url']
+
+    def test_get_current_site_invalid_certificate_context_with_site_configuration(self):
+        """
+        Test method returns correct data with site configuration.
+        """
+        with mock.patch('openedx.features.edly.utils.get_current_site_configuration') as current_site_configuration:
+            test_default_certificate_html_configurations = {
+                'default': {
+                    'accomplishment_class_append': 'accomplishment-certificate',
+                    'platform_name': 'fake-platform-name',
+                    'logo_src': 'fake-logo-path',
+                    'logo_url': 'fake-logo-url',
+                    'company_verified_certificate_url': 'fake-verified-certificate-url',
+                    'company_privacy_url': 'fake-privacy-url',
+                    'company_tos_url': 'fake-tos-url',
+                    'company_about_url': 'fake-about-url'
+                }
+            }
+
+            current_site_configuration.return_value = SiteConfigurationFactory(
+                site=self.request.site,
+                values={
+                    'platform_name': 'fake-2nd-platform-name',
+                    'MKTG_URLS': {
+                        'ROOT': 'fake-marketing-url',
+                        'PRIVACY': '/fake-privacy-path',
+                        'TOS_AND_HONOR': '/fake-tos-path',
+                    }
+                }
+            )
+            current_site_context_data = get_current_site_invalid_certificate_context(test_default_certificate_html_configurations)
+
+            expected_current_site_context = test_default_certificate_html_configurations['default']
+            expected_current_site_context['platform_name'] = self.request.site.configuration.values.get('platform_name')
+            assert expected_current_site_context['platform_name'] == current_site_context_data['platform_name']
+
+            marketing_urls = self.request.site.configuration.get_value('MKTG_URLS', {})
+            marketing_root_url = marketing_urls.get('ROOT')
+
+            tos_path = marketing_urls.get('TOS_AND_HONOR')
+            expected_company_tos_url = '{}{}'.format(marketing_root_url, tos_path)
+            assert expected_company_tos_url == current_site_context_data['company_tos_url']
```

Comment:
@zubair-arbi I'm unable to pass this check in test because `get_current_site()` from `get_tos_and_honor_code_url()` returns empty object. Can you suggest any other approach to it? I'm thinking about mocking `get_tos_and_honor_code_url()` function.

Reply:
Try using the first site (default site) for your test.

Reply:
Done